akka-cluster-tools/src/multi-jvm/scala/akka/cluster/client/ClusterClientSpec.scala

```diff
@@ -185,6 +293,52 @@ class ClusterClientSpec extends MultiNodeSpec(ClusterClientSpec) with STMultiNod
       enterBarrier("after-3")
     }
 
+    "report events" in within(15 seconds) {
+      runOn(client) {
+        implicit val timeout = Timeout(1.second.dilated)
+        val c = Await.result(system.actorSelection("/user/client").resolveOne(), timeout.duration)
+        val l = system.actorOf(Props(classOf[TestClientListener], c), "reporter-client-listener")
+
+        val expectedContacts = Set(first, second, third, fourth).map(node)
+        l ! TestClientListener.GetLatestContactPoints
+        fishForMessage() {
+          case contacts: Set[ActorSelection] @unchecked if contacts.map(_.anchorPath) == expectedContacts ⇒
+            true
+          case _ ⇒
+            l ! TestClientListener.GetLatestContactPoints
+            false
+        }
+      }
+
+      enterBarrier("reporter-client-listener-tested")
+
+      runOn(first, second, third) {
+        // Only run this test on a node that knows about our client. It could be that no node knows
+        // but there isn't a means of expressing that at least one of the nodes needs to pass the test.
+        implicit val timeout = Timeout(2.seconds.dilated)
+        val r = ClusterClientReceptionist(system).underlying
+        r ! GetClusterClients
+        val cps = expectMsgType[ClusterClients]
+        if (cps.clusterClients.exists(_.path.name == "client")) {
+          log.info("Testing that the receptionist has just one client")
```

Comment:
Does TestLatch support multi-jvm?

> On 12 May 2016, at 6:31 PM, Johan Andrén <notifications@github.com> wrote:
> 
> In akka-cluster-tools/src/multi-jvm/scala/akka/cluster/client/ClusterClientSpec.scala <https://github.com/akka/akka/pull/20462#discussion_r62982506>:
> 
> > +            l ! TestClientListener.GetLatestContactPoints
> > +            false
> > +        }
> > +      }
> > +
> > +      enterBarrier("reporter-client-listener-tested")
> > +
> > +      runOn(first, second, third) {
> > +        // Only run this test on a node that knows about our client. It could be that no node knows
> > +        // but there isn't a means of expressing that at least one of the nodes needs to pass the test.
> > +        implicit val timeout = Timeout(2.seconds.dilated)
> > +        val r = ClusterClientReceptionist(system).underlying
> > +        r ! GetClusterClients
> > +        val cps = expectMsgType[ClusterClients]
> > +        if (cps.clusterClients.exists(_.path.name == "client")) {
> > +          log.info("Testing that the receptionist has just one client")
> Could you use a TestLatch and fail after barrier unless at least one node has passed?
> 
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub <https://github.com/akka/akka/pull/20462/files/00cdee383ed7de56f67dabb5d91bda40806fb999#r62982506>